lib/extensions/ar_taggable.rb

```diff
@@ -74,7 +74,7 @@ def find_tags_by_grouping(list, options = {})
           next
         end
 
-        results.delete_if { |obj| !ret.include?(obj) }
+        results.reject { |obj| !ret.include?(obj) }
```

Comment:
reject returns a new array.  so you need to capture it:
```ruby
results = results.reject { |obj| !ret.include?(obj) }
```

although I agree that the double negative (`reject` and `!ret.include?`) is a bit hard to read.  Suggest to switch to:
```ruby
results = results.select { |obj| ret.include?(obj) }
```